fs/backgroundfetcher/background_fetcher.go

```diff
@@ -0,0 +1,167 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package backgroundfetcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/containerd/containerd/log"
+	"golang.org/x/time/rate"
+)
+
+type Option func(*BackgroundFetcher) error
+
+const (
+
+	// Specifies the maximum size of the work queue i.e., the maximum number
+	// of span managers that can be queued. In case of overflow, the `Add` call
+	// will block until a span manager is removed from the workqueue.
+	defaultWorkQueueSize = 100
```

Comment:
Does it have metrics/logs to tune the work queue size? Can we expose the queue length somewhere?

Reply:
Sure, I can make this configurable.